### ISSUE TITLE
Run security checks in CI

### ISSUE THREAD

**Issue by haimrait** [teletrace/teletrace#121]
**Is your feature request related to a problem? Please describe.**
Currently, we have security checks that not part of the CI process.

**Describe the solution you'd like**
Add an action/workflow or add a security section in a current CI workflow

**Describe alternatives you've considered**
blocked by #120
